netket/experimental/operator/_fermion_operator_2nd.py

```diff
@@ -474,24 +476,24 @@ def _op__imatmul__(self, other):
                 f"to operator with dtype {self.dtype}"
             )
 
-        terms = []
-        weights = []
+        new_operators = {}
         for t, w in self._operators.items():
             for to, wo in other._operators.items():
-                terms.append(tuple(t) + tuple(to))
-                weights.append(w * wo)
+                # if the last operator of t and the first of to are
+                # equal, we have a ĉᵢĉᵢ which is null.
+                if t[-1] != to[0]:
```

Comment:
@jwnys we agree right?

Reply:
can we add a test? ideally before merging into master...

Reply:
Yeah, but all tests pass, and even new ones.